unblob/handlers/archive/cpio.py

```diff
@@ -53,6 +80,20 @@ def calculate_chunk(
                     offset += self._pad_content(header, c_filesize, c_namesize)
                     break
 
+            c_mode = self._calculate_mode(header)
+
+            file_type = c_mode & 0o770000
+            sticky_bit = c_mode & 0o7000
+            permission = c_mode & 0o777
+
+            # heuristics 3: check mode field
+            if not (
+                any([file_type == ft for ft in C_FILE_TYPES])
+                and any([sticky_bit == sb for sb in C_STICKY_BITS])
+                and permission <= 0o777
+            ):
+                return
+
```

Comment:
Previously these checks had run only for `BinaryHandler` - is it also a fix for other CPIO Handlers?

Reply:
yes, all of them should be checked, it was a miss to check only for Binary only. Commit message also mentions this.